File: app/src/main/java/com/katkov/lolachievements/di/provider/ServerNameInterceptorProvider.java
package com.katkov.lolachievements.di.provider;

import com.katkov.lolachievements.data.cloud.intercepter.ServerNameInterceptor;
import com.katkov.lolachievements.data.local.prefser.EntryInfoHolder;

import javax.inject.Inject;
import javax.inject.Provider;

import okhttp3.Interceptor;

public final class ServerNameInterceptorProvider implements Provider<Interceptor> {

    private final EntryInfoHolder entryInfoHolder;

    @Inject
    ServerNameInterceptorProvider(EntryInfoHolder entryInfoHolder) {
        this.entryInfoHolder = entryInfoHolder;
    }

    @Override
    public Interceptor get() {
        return new ServerNameInterceptor(entryInfoHolder);
    }
}

File: app/src/main/java/com/katkov/lolachievements/di/provider/ServerNameInterceptorProvider.kt
package com.katkov.lolachievements.di.provider

import com.katkov.lolachievements.data.cloud.intercepter.ServerNameInterceptor
import com.katkov.lolachievements.data.local.prefser.LoginModelHolder
import okhttp3.Interceptor
import javax.inject.Inject
import javax.inject.Provider

class ServerNameInterceptorProvider
@Inject
internal constructor(private val loginModelHolder: LoginModelHolder) : Provider<Interceptor> {

    override fun get(): Interceptor = ServerNameInterceptor(loginModelHolder)
}
